packages/react-cosmos-component-playground/src/components/Tree/index.js

```diff
@@ -0,0 +1,207 @@
+      </div>
+      {node.expanded && (
+        <Tree
+          nodeArray={node.children}
+          onToggle={onToggle}
+          onSelect={onSelect}
+          selected={selected}
+          nestingLevel={nestingLevel + 1}
+          searchText={searchText}
+        />
+      )}
+    </div>
+  );
+};
+
+const nodeIsSelected = (node, selected) => {
+  if (!node.urlParams) {
+    return false;
+  }
+  return (
+    node.urlParams.component === selected.component &&
+    node.urlParams.fixture === selected.fixture
+  );
+};
+
+const TreeItem = ({ node, onSelect, isSelected, nestingLevel, searchText }) => {
+  const fixtureClassNames = classNames(styles.fixture, {
+    [styles.fixtureSelected]: isSelected
+  });
+
+  return (
+    <a
+      className={fixtureClassNames}
+      style={{
+        paddingLeft:
+          CONTAINER_LEFT_PADDING + (1 + nestingLevel) * INDENT_PADDING
+      }}
+      onClick={e => {
+        e.preventDefault();
+        e.stopPropagation();
+        onSelect(node);
+      }}
+    >
+      <FuzzyHighligher searchText={searchText} textToHighlight={node.name} />
+    </a>
+  );
+};
+
+class Tree extends React.Component {
+  componentDidMount() {
+    const el = this.selectedItem;
+    // scrollIntoView doesn't seem to exist in Jest/jsdom
+    if (el && el.scrollIntoView) {
+      el.scrollIntoView({
+        behavior: 'smooth'
+      });
+    }
+  }
+
+  render() {
+    const {
+      nodeArray,
+      onSelect,
+      onToggle,
+      selected,
+      searchText,
+      nestingLevel = 0
+    } = this.props;
+    const treeStyle = {};
+    if (process.env.NODE_ENV === 'development') {
+      treeStyle.backgroundColor = 'black';
+    }
+    return (
+      <div style={treeStyle}>
+        {nodeArray.map((node, index) => {
+          if (node.children) {
+            return (
+              <TreeFolder
+                key={index}
+                node={node}
+                onToggle={onToggle}
+                onSelect={onSelect}
+                selected={selected}
+                nestingLevel={nestingLevel}
+                searchText={searchText}
+              />
+            );
+          }
+          const isSelected = nodeIsSelected(node, selected);
+          return (
+            <div
+              ref={el => {
+                if (isSelected) {
+                  this.selectedItem = el;
+                }
+              }}
+              key={index}
+            >
+              <TreeItem
+                node={node}
+                onSelect={onSelect}
+                isSelected={isSelected}
+                nestingLevel={nestingLevel}
+                searchText={searchText}
+              />
+            </div>
+          );
+        })}
+      </div>
+    );
+  }
+}
+
+const nodeShape = shape({
+  name: string.isRequired,
+  expanded: bool,
+  urlParams: shape({
+    component: string.isRequired,
+    fixture: string.isRequired
+  })
+});
+nodeShape.children = arrayOf(nodeShape);
```

Comment:
Took me a few seconds to realize why you put this here 😅

Reply:
👍